# Создайте собственный класс-исключение, обрабатывающий ситуацию деления на нуль.
# Проверьте его работу на данных, вводимых пользователем.
# При вводе пользователем нуля в качестве делителя программа должна корректно
# обработать эту ситуацию и не завершиться с ошибкой.
def bad_func(*args):
    for i in args:
        try:
            print(1/i)
        except Exception as e:
            print('делить на ноль нельзя')
try:
    b=0
    while b == 0:
        a = bad_func(int(input('деление 1/ введите число -')))
except Exception as e:
    print('все накрылось')

print('программа закончила работать')

